integration-tests/bats/sql-pull.bats

```diff
@@ -459,6 +459,27 @@ teardown() {
     [[ ! "$output" =~ "v3" ]] || false
 }
 
+@test "sql-pull: dolt_pull with remote and remote ref" {
+    cd repo1
+    dolt checkout feature
+    dolt checkout -b newbranch
+    run dolt sql -q "show tables"
+    [ "$status" -eq 0 ]
+    [[ ! "$output" =~ "t1" ]] || false
+
+    # Explicitly specifying the remote and remote ref will merge in that branch
+    run dolt sql -q "call dolt_pull('origin', 'main');"
+    [ "$status" -eq 0 ]
+    run dolt sql -q "show tables"
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "t1" ]] || false
+
+    # Specifying a non-existent remote ref gives an error
```

Comment:
suggested additional tests off the top of my head, your discretion ofc all of this looks good already
- pull a feature branch that exists on the remote, but not locally
- pull a refspec branch of the form `refs/remotes/origin/feature`
- pull a branch that is not the active branch but exists locally
- pull a remote branch with active changes in the working set (no conflicts)
- pull branch with conflicting changes in the working set (probably same as above in Git)
- pull a branch that ff-merges with active/non-active branch
- pull a branch that can't merge with active/non-active branch

Reply:
Nice test case brainstorming! Some of these won't work yet, e.g. non-branch name parameters, using pull to merge a local branch (doesn't look like we support the '.' syntax yet that Git uses for this mode). I like the suggestion to test with different merge conditions. I'll poke around with that and add a little more coverage.